packages/wdio-browserstack-service/src/accessibility-handler.ts

```diff
@@ -0,0 +1,343 @@
+        if (!this._accessibility || this._framework !== 'mocha') {
+            return
+        }
+
+        if (isBrowserstackSession(this._browser)) {
+            const shouldScanTest = shouldScanTestForAccessibility(suiteTitle, test.title, this._accessibilityOptions)
+            const testIdentifier = this.getIdentifier(test)
+            this._testMetadata[testIdentifier] = {
+                scanTestForAccessibility : shouldScanTest,
+                accessibilityScanStarted : true
+            }
+
+            if (this._browser && isAccessibilityAutomationSession(this._accessibility)) {
+                let pageOpen = true
+
+                try {
+                    const currentURL = await (this._browser as WebdriverIO.Browser).getUrl()
+                    const url = new URL(currentURL)
+                    pageOpen = url?.protocol === 'http:' || url?.protocol === 'https:'
+                } catch (e) {
+                    pageOpen = false
+                }
+
+                try {
+                    if (pageOpen) {
+                        if (shouldScanTest) {
+                            log.info('Setup for Accessibility testing has started. Automate test case execution will begin momentarily.')
+                            await (this._browser as WebdriverIO.Browser).executeAsync(`
+                                const callback = arguments[arguments.length - 1]
+                                const fn = () => {
+                                    window.addEventListener('A11Y_TAP_STARTED', fn2)
+                                    const e = new CustomEvent('A11Y_FORCE_START')
+                                    window.dispatchEvent(e)
+                                }
+                                const fn2 = () => {
+                                    window.removeEventListener('A11Y_TAP_STARTED', fn)
+                                    callback()
+                                }
+                                fn()
+                            `)
+                        } else {
+                            await (this._browser as WebdriverIO.Browser).execute(`
+                                const e = new CustomEvent('A11Y_FORCE_STOP')
+                                window.dispatchEvent(e)
+                            `)
+                        }
+                        this._testMetadata[testIdentifier].accessibilityScanStarted = shouldScanTest
+
+                        if (shouldScanTest) {
+                            log.info('Automate test case execution has started.')
+                        }
+                    }
+                } catch (error) {
+                    log.error(`Exception in starting accessibility automation scan for this test case ${error}`)
+                }
+            }
+        }
+    }
+
+    async afterTest (suiteTitle: string | undefined, test: Frameworks.Test) {
+        if (!this._accessibility || this._framework !== 'mocha') {
```

Comment:
```suggestion
        if (!this._accessibility || this._framework !== 'mocha' || !isBrowserstackSession(this._browser)) {
```

and remove the if statement in line 164